### ISSUE TITLE
llama2 - loss declines too slowly

### ISSUE THREAD

**Issue by YunzeSong** [facebookresearch/llama#578]
Hi everyone, 

I am fine-tuning the llama2, but the loss is declining very slowly, and I am a little confused about the reason. Prior to this, I had fine-tuned the llama1 and the loss dropped significantly at that time

The picture below is the loss decline curve I was fine-tuning the llama2. 

I hope scholars who have similar problems and know how to solve them can give me some suggestions.

![W B Chart 2023_7_28 23_20_34](https://github.com/facebookresearch/llama/assets/129425169/d80be3c3-c9bf-437a-9b42-556c13945769)

Thanks!!!!!!

**Comment by Jasonli1997** [comment]
Have you tried a higher learning rate? I'm noticing the slow loss decline myself here fine-tuning Llama2-7b, comparing to when I fine-tuned Falcon-7b...

**Comment by wangzhonghai** [comment]
Why am I training the LLAMA7B model to start training NaN

**Comment by sssszh** [comment]
@wangzhonghai 
Hi, have you solved this problem yet?

I found the same problem when trying to peft fine-tune CodeLLama-7B (using LlamaForSequenceClassification), the Loss is always Nan during the fine-tuning.

Thanks！

**Comment by tiankanghui** [comment]
> [@wangzhonghai](https://github.com/wangzhonghai) Hi, have you solved this problem yet?
> 
> I found the same problem when trying to peft fine-tune CodeLLama-7B (using LlamaForSequenceClassification), the Loss is always Nan during the fine-tuning.
> 
> Thanks！
Hi, I also meet this nan error, and it occurs in different iterations (some times in 10, 40 and even in 5500!). I do not konw how to solve this problem. Do you have any suggestions?

Thank you!!
